packages/story-editor/src/app/media/utils/useProcessMedia.js

```diff
@@ -567,12 +571,38 @@ function useProcessMedia({
     ]
   );
 
+  /**
+   * Segment video using FFmpeg.
+   *
+   * @param {import('@googleforcreators/media').Resource} resource Resource object.
+   * @param {Function} handleSegmented function to handle segmentedFile processing.
+   */
+  const segmentVideo = useCallback(
+    async ({ resource: oldResource, segmentTime }, handleSegmented) => {
+      const { src: url, mimeType } = oldResource;
+      const segmentedFiles = await ffSegmentVideo(
+        await fetchRemoteFile(url, mimeType),
+        segmentTime
+      );
+
+      const addElement = handleSegmented(segmentedFiles);
+
+      for (const file of segmentedFiles) {
```

Comment:
Instead of a for loop, can we use `Promise.all` here.

Reply:
If we use `Promise.all` the files could get out of order correct?

Right now the segmentedFiles are ordered ---the loop calls addElement which adds the new page / element  --- based on the order of the array.

Reply:
Is there no to upload the files the add the elements in the correct order?

Reply:
Outside of the order of the array --- it would be possible to check the filenames as they get appended with 001, 002 etc... but we would need logic to add pages / elements between items i.e. file 4 is ready but 2, 3 haven't finished so we need to insert pages 2,3,4 to add the `4` element.


I think it ends up being cleaner / simpler to use the array order.

Reply:
As noted [here](https://github.com/GoogleForCreators/web-stories-wp/pull/12335#issuecomment-1254836974), I think it might be best to upload all the video first before creating the pages. What if one or more of the videos fail to upload, that could make a mess, where page 2 of 5 is missing.

Reply:
`uploadMedia` accepts multiple files.

So instead of this for loop we can just do `await uploadMedia(segmentedFiles, { onUploadSuccess });`

Reply:
Can follow up this conversation here:

https://github.com/GoogleForCreators/web-stories-wp/pull/12335#discussion_r978162303